service/src/test/java/cz/muni/fi/pa165/service/ManufacturerServiceTest.java

```diff
@@ -0,0 +1,207 @@
+package cz.muni.fi.pa165.service;
+
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+
+import cz.muni.fi.pa165.AbstractServiceTest;
+import cz.muni.fi.pa165.dao.ManufacturerDAO;
+import cz.muni.fi.pa165.entity.Bottle;
+import cz.muni.fi.pa165.entity.Manufacturer;
+import org.hibernate.service.spi.ServiceException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.DataIntegrityViolationException;
+
+import javax.validation.ConstraintViolationException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author Jakub Fiser
+ * @date 24/11/2016
+ */
+public class ManufacturerServiceTest extends AbstractServiceTest {
+
+    @Mock
+    private ManufacturerDAO manufacturerDAO;
+
+    @Mock
+    private BottleService bottleService;
+
+    @Autowired
+    @InjectMocks
+    private ManufacturerService manufacturerService;
+
+    private Manufacturer manufacturer1;
+    private Manufacturer manufacturer2;
+    private Bottle bottle1;
+    private Bottle bottle2;
+    private Date date = new Date();
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+
+        manufacturer1 = new Manufacturer();
+        manufacturer1.setId(1L);
+        manufacturer1.setName("Manufacturer1");
+
+        manufacturer2 = new Manufacturer();
+        manufacturer2.setId(2L);
+        manufacturer2.setName("Manufacturer2");
+
+        bottle1 = new Bottle();
+        bottle1.setId(1L);
+        bottle1.setStickerID("id1");
+        bottle1.setProduced(date);
+        bottle1.setToxic(false);
+
+        bottle2 = new Bottle();
+        bottle2.setId(2L);
+        bottle2.setStickerID("id2");
+        bottle2.setProduced(date);
+        bottle2.setToxic(false);
+    }
+
+    @Test
+    public void createManufacturer() {
+        manufacturerService.createManufacturer(manufacturer1);
+        verify(manufacturerDAO).save(manufacturer1);
+    }
+
+    @Test(expected = ConstraintViolationException.class)
```

Comment:
Jeste jsem se chtel zeptat na tuhle vyjimku. Ze zadani mame pouzivat DataAccessException nebo jeji podtridu, coz ConstraintViolationException neni. Mame to v service kontrolovat a vyhodit podtridu DataAccessException, nebo treba ServiceException, nebo to nechame?